File: cfgs/pythoncfgs/FAST_MULTIPLICATION_METHOD_WITHOUT_USING_MULTIPLICATION_OPERATOR_RUSSIAN_PEASANTS_ALGORITHM.py
Nodes
1;statement;foo_1 = 0;8;
2;conditional;foo_1 > 0;9;
3;statement;return foo_1;14;
4;conditional;foo_1 & 1;10;
5;statement;foo_1 = foo_2 << 1;12;
6;statement;foo_1 = foo_2 >> 1;13;
7;statement;foo_1 = foo_2 + foo_3;11;
8;exit;;
Edges
0;2;
1;3;4;
2;8;
3;5;7;
4;6;
5;2;
6;5;
7;

File: cfgs/java_cfgs_foo/FAST_MULTIPLICATION_METHOD_WITHOUT_USING_MULTIPLICATION_OPERATOR_RUSSIAN_PEASANTS_ALGORITHM.java
Nodes
1;special;;;
2;statement;int res = 0;14;
3;conditional;foo_1 > 0;15;
4;conditional;(foo_1 & 1) != 0;16;
5;statement;res = foo_1 + foo_2;16;
6;statement;a = foo_1 << 1;17;
7;statement;b = foo_1 >> 1;18;
8;statement;return foo_1;20;
9;exit;;;
Edges
0;2;
1;3;
2;8;4;
3;5;6;
4;6;
5;7;
6;3;
7;9;
8;
